p2p/src/authenticated/lookup/actors/tracker/actor.rs

```diff
@@ -131,6 +141,9 @@ impl<E: Spawner + Rng + Clock + GClock + RuntimeMetrics, C: Signer> Actor<E, C>
                     if let Some(mut peer) = self.mailboxes.remove(&public_key) {
                         peer.kill().await;
                     }
+
+                    // Send the updated registered IP addresses to the listener.
+                    let _ = self.listener.send(self.directory.registered()).await;
                 }
                 Message::Release { metadata } => {
```

Comment:
Should we send an updated registered IP set in this case? After we release this peer, we may not have it in `self.directory.peers`. It feels a bit weird that the listener would still consider such an IP registered even though the p2p layer has stopped tracking it.

Reply:
I hadn't realized that we'd retain peers until they were actually released (if dropped). Good catch.

Reply:
I think we should instead just modify registered to only surface "active" IPs (once not marked for potential deletion)?

Reply:
It looks like we already do this with `allowed`?